java/runtime/src/main/java/org/ray/runtime/raylet/RayletClientImpl.java

```diff
@@ -172,7 +172,7 @@ private static ByteBuffer convertTaskSpecToFlatbuffer(TaskSpec task) {
     final int parentTaskIdOffset = fbb.createString(task.parentTaskId.toByteBuffer());
     final int parentCounter = task.parentCounter;
     final int actorCreateIdOffset = fbb.createString(task.actorCreationId.toByteBuffer());
-    final int actorCreateDummyIdOffset = fbb.createString(UniqueId.NIL.toByteBuffer());
+    final int actorCreateDummyIdOffset = fbb.createString(task.actorId.toByteBuffer());
```

Comment:
The dummy object ID isn't actually the actor ID, right? It's the dummy object for the actor creation task.

Reply:
In the Python frontend, that's true, but it turns out that Java is actually using the dummy object ID as the actor ID [here](https://github.com/ray-project/ray/blob/master/java/runtime/src/main/java/org/ray/runtime/AbstractRayRuntime.java#L222). I think it's actually nice to have some association like this.

Reply:
Oh I see, thanks.

Reply:
One thing I found useful about having this association is when I debug issues about actor reconstruction, I can easily find the actor creation task id from the actor id, and vice versa.